### PR TITLE
Refactor message_create_get function

### PR DESCRIPTION
**Pull request by paisoncodes**
This function only get messages now 
I added docstring and comments
I removed the POST part
I also changed the function name and the url path name to "message_get" and "get_message" respectively.

### CODE REVIEW

**Review by o4codes — CHANGES_REQUESTED** [review]
Nice work.
The only issue is  that with this PR, the send message endpoint has been removed.

The retrieve all messages and send message were combined into a single FBV since they both use the same path param. So sending a post request sends a message, sending a GET request retrieves all messages.

@paisoncodes

**Comment by zxenonx** [comment]
@o4codes  oh, i told him to remove the post part from this function. There's already a send_message inside the code right? The one you implemented.

**Comment by o4codes** [comment]
@zxenonx 
The post part was the send message.
It was joined into a single function(retrieve and post)

**Comment by zxenonx** [comment]
@o4codes oh. Sorry. If that's the isn't it better to create a MessageView cbv and implement get/post/delete/put inside it?
Because there's already a cbv class for message.

**Comment by o4codes** [comment]
All functions in a CBV has to share the same URL. So for messages, having a create, get all, get, delete, and update in a single class won't work. Because the url structure for send and retrieve all will be different from the rest.

What we can implement is two CBV. One CBV is already implemented by @blacdev and that's the CBV individual message (put, delete and get) because they are all share the same url structure.

The other CBV we can implement is a MessageCreateRetrieve class. It will have a function for get (which is to retrieve all messages) and post(create a new message) because those two endpoints share the same url structure.

Formerly we had a single FBV that handles the create and get all messages and this was done to work best with @austinoski decorator. 

So if refactoring to a class based view for that function will be done,  the decorator will have to be removed.

**Comment by paisoncodes** [comment]
Okay so what should be done?? @zxenonx @o4codes

**Comment by o4codes** [comment]
The message_create_get FBV can be turned to a class based view, but you'd have to do away with the decorator

@paisoncodes

**Comment by paisoncodes** [comment]
Okay, but @zxenonx told me someonenwas working on the post method already.

@o4codes

**Comment by o4codes** [comment]
@paisoncodes 
Please go and ahead and work on the CBV for post and retrieve all messages. You already have the FBV just transform it to a CBV. Time is of the essence.

**Comment by zxenonx** [comment]
@paisoncodes sorry for the inconvenience

**Comment by zxenonx** [comment]
Good work @paisoncodes

**Review by zxenonx — APPROVED** [review]

**Comment by zxenonx** [comment]
@o4codes & @man-of-mind

File: backend/messaging.py [hunk 1]
```diff
@@ -34,35 +34,59 @@
...
         date = request.GET.get("date", None)
+
+        # Serialize the request data i.e convert it to json
         params_serializer = GetMessageSerializer(data=request.GET.dict())
+
+        # check if the serialized params are valid. If true, run request.
         if params_serializer.is_valid():
+
+            # Check to see if the room exists in the database
             room = DB.read_query("dm_rooms", query={"_id": room_id})
```
Comment (man-of-mind) [inline backend/messaging.py:69]: Hi boss, you need to set the organization id to be equal to the org_id passed in the url, right now, DB.read_query will bs using the hardcoded org_id in db.py file. So before using DB.read_query, u can do sth like DB.organization_id = org_id

File: backend/messaging.py [hunk 2]
```diff
@@ -71,27 +95,72 @@ def message_create_get(request, room_id):
...
         serializer = MessageSerializer(data=request.data)
 
+        # check if the serialized data is valid
         if serializer.is_valid():
+            # if true continue
             data = serializer.data
             room_id = data["room_id"]  # room id gotten from client request
 
+            # Check to see if the room exists in the database
             room = DB.read_query("dm_rooms", query={"_id": room_id})
```
Comment (man-of-mind) [inline backend/messaging.py:156]: U already instantiated the DataStorage class as data_storage, u dont need to be using DB.read_query instead data_stotage.read_query

**Review by o4codes — APPROVED** [review]
Good work @paisoncodes

File: backend/messaging.py (hunk 1 above)
Comment (zxenonx) [reply backend/messaging.py:69]: Oh, true. @paisoncodes

File: backend/messaging.py (hunk 2 above)
Comment (zxenonx) [reply backend/messaging.py:156]: @paisoncodes. Or you can also just remove the data_storage you instantiated and use what @man-of-mind suggested above.

File: backend/messaging.py (hunk 1 above)
Comment (paisoncodes) [reply backend/messaging.py:69]: I already did that when I did data_storage. I'll just change the DB to data_storage

Comment (zxenonx) [reply backend/messaging.py:69]: I think we should keep DB object. It's already an instance, so we don't need to instantiate again before assigning the org_id to it. @paisoncodes

Comment (man-of-mind) [reply backend/messaging.py:69]: @paisoncodes u didn't make the change here.

Comment (paisoncodes) [reply backend/messaging.py:69]: It's done now... @man-of-mind

Comment (man-of-mind) [reply backend/messaging.py:69]: @paisoncodes You didn't push your changes?
Seems it's still the same, there is no place you set the ord_id in the get method, u only did it for the post method.

Comment (paisoncodes) [reply backend/messaging.py:70]: pushed. It's in the commit below

**Review by man-of-mind — APPROVED** [review]
Well done boss @paisoncodes

**Comment by zxenonx** [comment]
> Well done boss @paisoncodes

It's good now @paisoncodes. Thanks

**Pull request merged**